Automattic/wp-calypso
# Add checkbox to accept third party developers account creation

#### Proposed Changes

* This PR adds a required checkbox to the checkout process that is displayed if a _Marketplace_ product is in the cart. In that case, the Submit button does not submit the purchase until the user checks the checkbox.

Further context can be found here: pdt2If-e4-p2 and in the following comment: pdt2If-e4-p2#comment-150 and the final result in this comment pdt2If-e4-p2#comment-160

|Before Changes | After Changes Un-checked|After Changes Un-checked clicking the `Pay` button|  After Changes Checked|
|-------|------|------|-------|
|![CleanShot 2022-07-20 at 11 31 50@2x](https://user-images.githubusercontent.com/3519124/179949178-60bd553f-f902-44b4-be9f-4631ca261698.png)|    ![CleanShot 2022-07-26 at 08 42 25@2x](https://user-images.githubusercontent.com/3519124/180941119-8d4973cc-5338-4e4a-8ab8-5667f033b1d3.png)       |      ![CleanShot 2022-07-26 at 08 46 33@2x](https://user-images.githubusercontent.com/3519124/180941465-1b3b87ce-968b-4167-921c-68795899c203.png)    |     ![CleanShot 2022-07-26 at 08 47 53@2x](https://user-images.githubusercontent.com/3519124/180941647-bafb530d-8eba-48b0-b4a5-497b7a62ad88.png)     |

#### Testing Instructions



##### Test 1

1. Navigate to the plugins page of a site that allows buying Plugins (Pro, Business or eCommerce)
2. Click on a Marketplace plugin, for example `/plugins/automatewoo/:siteId`
3. Click on `Purchase and activate`
4. Ensure there is a checkbox on top of the `Pay` button
5. Click on the `Pay` button
7. Ensure the checkout process is not completed, the checkbox above is highlighted in blue (a blue border is added to the checkbox) and there is an error message indicating that `The terms above need to be accepted`
8. Check the checkbox
9. Ensure the blue highlighting border is removed and the error message disappears.
10. Un-check the checkbox again
11. Step 7 again, highlighting in blue and error message is displayed
12. Check the checkbox 
13. Step 9 again, highlighting and error message disappear.
14. Click on `Pay` and make sure the plugin is installed as expected.

##### Test 2

1. Navigate to a Site and Upgrade the Plan, for example from Business to eCommerce, e.g. `plans/:siteId`
2. Ensure there is not any checkbox and the `Pay` button is enabled when the required data is populated
3. Go back or select the top-left X button but when prompted select `Leave items` to leave the plan upgrade in the cart
4. Repeat **Test 1** above. Ensure that with both products, a Plan Upgrade and a Marketplace item in the Cart, the checkbox is working as expected.
5. Remove the Marketplace plugin from the cart.
6. Ensure the checkbox is **not** displayed and the `Pay` button works as expected.

---

Please @Automattic/editorial , could you please review the wording of the messages? 

The following messages have been added:

In the checkbox label:
> You agree that an account may be created on a third party developer’s site related to the products you have purchased.

And the following message in the error message when submitting with checkbox un-checked:
> The terms above need to be accepted


Related to #64799
